apps/comps/e2e/utils/test-helpers.ts

```diff
@@ -119,16 +120,24 @@ export async function createPrivyAuthenticatedRpcClient(params: {
   // Create RPC client with Privy auth
   const rpcClient = await createTestRpcClient(privyToken);
 
-  // Login to create/update user
+  // Login to create user
   let user = await rpcClient.user.login();
 
   // Update name if provided
   if (params.userName) {
-    user = await rpcClient.user.updateProfile({ name: params.userName });
+    user = await rpcClient.user.updateProfile({
+      name: params.userName,
+    });
   }
 
-  // Add a small delay to ensure session is properly saved
-  await new Promise((resolve) => setTimeout(resolve, 100));
+  // Link custom wallet if provided. Note: we call `linkWallet` for test convenience. In reality,
+  // creating a user and linking a wallet are two separate operations.
+  if (params.walletAddress && params.walletAddress !== testEmbeddedWallet) {
+    MockPrivyClient.linkWallet(uniquePrivyId, params.walletAddress);
```

Comment:
this change has to do with the way custom wallet linking is handled. before, there were edge cases where our _login_ flow could add a custom linked wallet to the user's account.

we now remove this from our login flow logic, and only let explicit `linkWallet` calls handle this feature. hence, this helper just makes it easy to do both in one fell swoop.